### ISSUE TITLE
RE-open: Blazor redirect to cshtml not work with renderMode = Server 

### ISSUE THREAD

**Issue by andronachev** [dotnet/aspnetcore#39391]
Hello,

I would like to reopen this issue as I don't know why but it has been closed and marked as resolved but within the comments there is no fix: 

 https://github.com/dotnet/aspnetcore/issues/18849

We have the exact same issue: render-mode="Server", within OnItializedAsync we use the NavigationManager to navigate to the Login Screen: 

`
	navigationManager.NavigateTo("/identity/account/login" + (string.IsNullOrWhiteSpace(returnUrl) ? string.Empty : $"?returnUrl=/{returnUrl}"));

 `

Nothing happens unless we use "forceLoad: true", which I would like to avoid or at least understand why do we have to use it. And even then, somehow the lifecycle of the application has changed, because if render-mode="Server", the debugger will go past the navigationManager.NavigateTo line of code and continue execution, so, we also have to add a blank "return;" after NavigateTo, if we use render-mode="Server". 

With our previous render-mode= ServerPrerendered, the debugger would stop at "NavigateTo" and perform the navigation, without "forceLoad: true". 
 
Tl;dr: 

1. render-mode="ServerPrerendered" : the NavigateTo method works within OnInitializedAsync, works as expected, the debugger breaks execution and the page is redirected etc 
2. render-mode="Server" (we just did this change): the NavigateTo method does not work anymore as-is, and it only seems to work with "forceLoad: true", however, the lifecycle is somehow changed because the debugger does not break at NavigateTo, and so we have to add a blank "return;" to stop the execution ourselves

**Comment by weinzii** [comment]
I have the same issue. Have you found a solution for it? @andronachev
